includes/Context.php

```diff
@@ -357,14 +357,25 @@ public function get_amp_mode() {
 			return false;
 		}
 
-		$exposes_support_mode = method_exists( 'AMP_Theme_Support', 'get_support_mode' )
-			&& defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
-			&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
-			&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
+		$amp_plugin_version_2_or_higher = version_compare( AMP__VERSION, '2.0.0', '>=' );
+
+		if ( $amp_plugin_version_2_or_higher ) {
+			$exposes_support_mode = method_exists( 'AMP_Options_Manager', 'get_option' )
+				&& defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
+				&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
+				&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
+		} else {
+			$exposes_support_mode = method_exists( 'AMP_Theme_Support', 'get_support_mode' )
+				&& defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
+				&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
+				&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
+		}
```

Comment:
Maybe the checks for the three constants would be better moved out of the if-else, into a separate variable since they're exactly the same.

Reply:
I thought that too but wasn't sure if it'd be worth it. Two of us thinking the same thing says it's probably a good idea, so I'll do that 🙂